Add a section on configuring the Runtime to Runtime Agent docs (#9041)
kyma-project/kyma: docs/runtime-agent/03-02-configuring-runtime.md

---
title: Configuring the Runtime
type: Details
---

> **NOTE:** To represent API and Event Definitions of the connected Applications on Runtime, Open Service Broker API usage is recommended.

In Kyma Runtime, during Runtime configuration, Application's Packages are integrated into the [Service Catalog](components/service-catalog) using [Application](components/application-connector#custom-resource-application) custom resources and [Application Broker](components/application-connector#architecture-application-broker). By default, a single Application is represented as a [ServiceClass](components/service-catalog/#architecture-resources), and a single Package is represented as a [ServicePlan](components/service-catalog/#architecture-resources) in the Service Catalog. Read more about [API Packages](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-packages-api.md) document.

Runtime Agent periodically requests for the configuration of its Runtime from Compass. Changes in the configuration for the Runtime are applied by the Runtime Agent on the Runtime.

> To fetch the Runtime configuration, Runtime Agent calls the [`applicationsForRuntime`](https://github.com/kyma-incubator/compass/blob/master/components/director/pkg/graphql/schema.graphql) query offered by the Compass component called Director. The response for the query contains a page with the list of Applications assigned for the Runtime and info about the next page. Each Application will contain only credentials that are valid for the Runtime that called the query. Each Runtime Agent can fetch the configurations for Runtimes that belong to its tenant, there is no validation if the Runtime Agent is fetching the configuration for the Runtime on which it runs.

Reviewer: > The response for the query contains a page with the list of Applications assigned for the Runtime and info about the next page.

I am not sure if it is not going to much into detail. Maybe it is enough to say that it contains Applications assigned for the Runtime?

> To fetch the Runtime configuration, Runtime Agent calls the [`applicationsForRuntime`](https://github.com/kyma-incubator/compass/blob/master/components/director/pkg/graphql/schema.graphql) query offered by the Compass component called Director. The response for the query contains a page with the list of Applications assigned for the Runtime and info about the next page. Each Application will contain only credentials that are valid for the Runtime that called the query. Each Runtime Agent can fetch the configurations for Runtimes that belong to its tenant, there is no validation if the Runtime Agent is fetching the configuration for the Runtime on which it runs.

Reviewer: > Each Runtime Agent can fetch the configurations for Runtimes that belong to its tenant, there is no validation if the Runtime Agent is fetching the configuration for the Runtime on which it runs.

I am not exactly sure if that is true as the certificate is generated for a specific System Auth which represents the specific Runtime. I believe we were planning to have such validation. Are you sure it is not in place?

Reviewer 2: Yes, exactly, we're planning to have it, but not sure when exactly. I know that the No 1. priority is to have Application separation (App can only see and modify itself within the tenant).

---
title: Configuring the Runtime
type: Details
---

> > **NOTE:** To represent API and Event Definitions of the connected Applications on Runtime, Open Service Broker API usage is recommended.

Reviewer 3: ```suggestion
> **NOTE:** To represent API and Event definitions of the Applications connected to a Runtime, Open Service Broker API usage is recommended.
```

Reviewer 3: I am not sure about this one, though 🤔

Author: I've consulted a dev about it. Apart from `Event Definitions`, the suggestion is fine 👍

> In Kyma Runtime, during Runtime configuration, Application's Packages are integrated into the [Service Catalog](components/service-catalog) using [Application](components/application-connector#custom-resource-application) custom resources and [Application Broker](components/application-connector#architecture-application-broker).

Author: Save for the `the`, good catch! 👍

> By default, a single Application is represented as a [ServiceClass](components/service-catalog/#architecture-resources), and a single Package is represented as a [ServicePlan](components/service-catalog/#architecture-resources) in the Service Catalog.

Reviewer 3: ```suggestion
By default, a single Application is represented as a [ServiceClass](components/service-catalog/#architecture-resources), and a single Package is represented as a [ServicePlan](components/service-catalog/#architecture-resources) in Service Catalog.
```

Author: Good catch! Thanks!

> Read more about [API Packages](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-packages-api.md) document.

Reviewer 3: ```suggestion
Refer to the documentation to learn more about [API Packages](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-packages-api.md).
```

Runtime Agent periodically requests for the configuration of its Runtime from Compass.

> Changes in the configuration for the Runtime are applied by the Runtime Agent on the Runtime.

Reviewer 3: ```suggestion
Changes in the Runtime configuration are applied by the Runtime Agent on the Runtime.
```

Author: As discussed, I'm afraid this change does not retain the original meaning. I'll go with the original.

To fetch the Runtime configuration, Runtime Agent calls the [`applicationsForRuntime`](https://github.com/kyma-incubator/compass/blob/master/components/director/pkg/graphql/schema.graphql) query offered by the Compass component called Director.
The response for the query contains Applications assigned for the Runtime.
Each Application contains only credentials that are valid for the Runtime that called the query.
Each Runtime Agent can fetch the configurations for Runtimes that belong to its tenant.

> Runtime Agent reports back to the Director the Runtime-specific [LabelDefinitions](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-02-labels.md#labeldefinitions) that represent Runtime configuration together with their values.

Reviewer 3: ```suggestion
Runtime Agent reports back to the Director the Runtime-specific [LabelDefinitions](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-02-labels.md#labeldefinitions), that represent Runtime configuration, together with their values.
```

Author: I know what you were trying to do here, but I'm not sure you can put this subordinate `that` clause in commas like that.

> 

Reviewer 3: ```suggestion
```

> Runtime-specific LabelDefinitions:

Reviewer 3: ```suggestion
Runtime-specific LabelDefinitions are Events Gateway URL and Runtime Console URL.
```

Author: I guess it makes more sense when there are only two of them 👍
